src/server/api/routes/nft/metadata.js

```diff
@@ -0,0 +1,165 @@
+const axios = require('axios');
+const BigNumber = require('bignumber.js');
+const { Router } = require('express');
+
+const { handleRestfulError } = require('../../middleware/error');
+
+const {
+  LIKECOIN_CHAIN_API,
+  LIKECOIN_API_BASE,
+} = require('../../../config/config');
+
+const router = Router();
+
+router.get('/nft/metadata', async (req, res, next) => {
+  try {
+    const { class_id: classId } = req.query;
+    if (!classId) {
+      res.status(400).send('MISSING_CLASS_ID');
+      return;
+    }
+
+    const [
+      [classData, iscnData],
+      ownerInfoRes,
+      listingInfoRes,
+      purchaseInfoRes,
+    ] = await Promise.all([
+      getNFTClassAndISCNMetadata(classId),
+      axios.get(
+        `${LIKECOIN_CHAIN_API}/likechain/likenft/v1/owner?class_id=${classId}`
+      ),
+      axios.get(
+        `${LIKECOIN_CHAIN_API}/likechain/likenft/v1/listings/${classId}`
+      ),
+      axios
+        .get(`${LIKECOIN_API_BASE}/likernft/purchase?class_id=${classId}`)
+        .catch(err => {
+          // skip 404 error for non Writing NFT
+          if (err.response?.status === 404) {
+            return null;
+          }
+          throw err;
+        }),
+    ]);
+
+    const owners = ownerInfoRes.data?.owners || [];
+    const listingInput = listingInfoRes.data?.listing || [];
+    const listings = formatAndFilterListing(listingInput, owners);
+    const purchaseInfo = purchaseInfoRes?.data;
+
+    res.json({
+      classData,
+      iscnData,
+      owners,
+      listings,
+      purchaseInfo,
+    });
+  } catch (err) {
+    if (err.message === 'NFT_CLASS_NOT_FOUND') {
+      res.status(404).send(err.message);
+    } else {
+      handleRestfulError(req, res, next, err);
+    }
+  }
+});
+
+async function getNFTClassAndISCNMetadata(classId) {
+  const {
+    data: { class: chainMetadata },
+  } = await axios
+    .get(`${LIKECOIN_CHAIN_API}/cosmos/nft/v1beta1/classes/${classId}`)
```

Comment:
Should use [getChainNFTClassMetadataEndpoint](https://github.com/nwingt/liker-land/blob/beee54060ac33e3b91d4d96cee1664177e194bd7/src/util/api/index.js#L140)

Reply:
Seems we would get a `SyntaxError: Cannot use import statement outside a module` error.
Since we're still using the `common JS` type for server, that way we cannot share functions/configs across the repo...
@williamchong do you have some insight on this issue?

Reply:
Firstly we should not require any file outside of `server` folder from any file inside `server` since it causes error on build
Maybe we have have a file in server for apis for now, and figure out the share part later